File: test_feature/planning/api/src/main/java/com/github/klee0kai/stone/test_feature/planning/project/Project.java
package com.github.klee0kai.stone.test_feature.planning.project;

import com.github.klee0kai.stone.test_feature.planning.model.ProjectInfo;
import com.github.klee0kai.stone.test_feature.planning.model.TaskInfo;

public interface Project {

    String getId();

    ProjectInfo projectInfo();

    void addTask(TaskInfo taskInfo);

    TaskInfo findTask(String taskId);

}

File: test_feature/planning/api/src/main/java/com/github/klee0kai/stone/test_feature/planning/project/Project.kt
package com.github.klee0kai.stone.test_feature.planning.project

import com.github.klee0kai.stone.test_feature.planning.model.ProjectInfo
import com.github.klee0kai.stone.test_feature.planning.model.TaskInfo

interface Project {

    val id: String?

    fun projectInfo(): ProjectInfo?

    fun addTask(taskInfo: TaskInfo?)

    fun findTask(taskId: String?): TaskInfo?
}
